### PR TITLE
Fix dispose ordering

### PR DESCRIPTION
**Pull request by reyang**
Looks like the exporter relies on the token, so if the token is Disposed before the exporter task finished it job, it is likely to cause exception?

### CODE REVIEW

File: src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs [hunk 1]
```diff
@@ -56,8 +56,8 @@ protected override void Dispose(bool disposing)
                 try
                 {
                     this.token.Cancel();
-                    this.token.Dispose();
                     this.exportTask.Wait();
+                    this.token.Dispose();
```
Comment (ThomsonTan) [inline src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs:60]: Dispose `this.exportTask` after `this.exportTask.Wait()`?

**Review by cijothomas — APPROVED** [review]

**Review by utpilla — APPROVED** [review]
This won't throw an exception though as `token.IsCancellationRequested` will still be `true` even after `token.Dispose()` is called.

File: src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs (hunk 1 above)
Comment (utpilla) [reply src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs:60]: This is interesting. I found this article by Stephen Toub on why we don't have to dispose tasks in most cases: https://devblogs.microsoft.com/pfxteam/do-i-need-to-dispose-of-tasks/

Comment (reyang) [reply src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs:60]: I think:
1. disposing the task is not the goal of this PR.
2. we need to refactor this class to replace task with explicit thread, there is already a comment https://github.com/open-telemetry/opentelemetry-dotnet/blob/bec00a9a928c296c04f9e0e65f26c2925441b575/src/OpenTelemetry/Metrics/PeriodicExportingMetricReader.cs#L33.

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**